File: Data Structures (Java & C++)/Project 3/Project 3 C++/Node.cpp
//
//  Node.cpp
//  Project 3 C++
//
//  Created by Alexandru Gutu on 10/18/14.
//  Copyright (c) 2014 Alexandru Gutu. All rights reserved.
//

#include <iostream>
#include "Node.h"

using namespace std;

// Node function definitions
Node::Node(string input)
{
	this->data = input;
	this->count = 1;
}

Node::Node(string input, Node* next)
{
	this->data = input;
	this->count = 1;
	this->next = next;
}

string Node::getData()
{
	return this->data;
}

Node* Node::getNext()
{
	return this->next;
}

int Node::getCounter()
{
	return this->count;
}

void Node::setData(string input)
{
	this->data = input;
}

void Node::setNext(Node* next)
{
	this->next = next;
}

void Node::increaseCount()
{
	this->count++;
}

File: Data Structures (Java & C++)/Project 3/Project 3 Java/Node.java
/**
 * Created by Alexandru Gutu on 10/18/14.
 * Copyright (c) 2014 Alexandru Gutu. All rights reserved.
 */

// Node prototype and definition

public class Node
	{
		private String data;
		private int counter;
		private Node next;

		public Node(String input)
			{
				data = input;
				counter = 1;
			}

		public Node(String input, Node next)
			{
				data = input;
				counter = 1;
				this.next = next;
			}

		public String getData()
			{
				return this.data;
			}

		public Node getNext()
			{
				return this.next;
			}

		public int getCounter()
			{
				return this.counter;
			}

		public void setData(String input)
			{
				this.data = input;
			}

		public void setNext(Node next)
			{
				this.next = next;
			}

		public void increaseCount()
			{
				this.counter++;
			}
	}
